### ISSUE TITLE
Custom SQL x-axis time shows very crowded

### ISSUE THREAD

**Issue by hengfeiyang** [openobserve/openobserve#635]
### Which ZincObserve functionalities are relevant/related to the feature request?

dashboards

### Description

<img width="798" alt="image" src="https://user-images.githubusercontent.com/1628250/234580164-e92c950e-9b97-4422-9c7e-60a3a51af434.png">

Like this chart, the time labels can't be format as time. i think the reason is when we use default x-axis the name is `_timestamp`, we handle it. But when we use `custom SQL`, we can't use `_timestamp` as label name, my custom SQL like this:

```sql
SELECT histogram(_timestamp, '1 minute') as "x_axis_1", count(status) as "y_axis_1"  FROM "default"  GROUP BY "x_axis_1" ORDER BY "x_axis_1"
```

After custom SQL, the x-axis can't be format as time.

### Proposed solution

I suggest, we add an option for x-axis `is a time field`, then will know it is a time field, we need to handle it use time format.

### Alternatives considered

no

**Comment by ktx-kirtan** [comment]
@prabhatsharma We need a way to identify the time field. The proposed solution makes sense as we do not have any other way to identify date fields. Other platforms might be able to identify date fields based on the field type. But we do not have any other way to identify the date field. We do have the config option where the date field is there, but if that comes with the custom SQL, we again do not know what fields are selected from the SQL. This will also help us fix #655, #532 and allow a natively supported way of time series.

**Comment by ktx-kirtan** [comment]
Moving this to next milestone
